internalshared/configutil/http_response_headers.go

```diff
@@ -0,0 +1,186 @@
+package configutil
+
+import (
+    "fmt"
+    "net/textproto"
+    "strconv"
+    "strings"
+)
+
+var DefaultHeaderNames = []string {
+    "Content-Security-Policy",
+    "X-XSS-Protection",
+    "X-Frame-Options",
+    "X-Content-Type-Options",
+    "Strict-Transport-Security",
+    "Content-Type",
+}
+
+var ValidCustomStatusCodeCollection = []string {
+    "default",
+    "1xx",
+    "2xx",
+    "3xx",
+    "4xx",
+    "5xx",
+}
+
+const (
+    contentSecurityPolicy = "default-src 'none';   connect-src 'self';  img-src 'self' data:; script-src 'self'; style-src 'unsafe-inline' 'self'; form-action  'none'; frame-ancestors 'none'; font-src 'self'"
+    xXssProtection = "1; mode=block"
+    xFrameOptions = "Deny"
+    xContentTypeOptions = "nosniff"
+    strictTransportSecurity = "max-age=31536000; includeSubDomains"
+    contentType = "text/plain; charset=utf-8"
+)
+
+func ParseDefaultHeaders(h string) string {
+    switch h {
+    case "Content-Security-Policy":
+        return contentSecurityPolicy
+    case "X-XSS-Protection":
+        return xXssProtection
+    case "X-Frame-Options":
+        return xFrameOptions
+    case "X-Content-Type-Options":
+        return xContentTypeOptions
+    case "Strict-Transport-Security":
+        return strictTransportSecurity
+    case "Content-Type":
+        return contentType
+    default:
+        return ""
+    }
+}
+
+func setDefaultResponseHeaders(c map[string]string) map[string]string {
+    defaults := make(map[string]string)
+    // adding all parsed default headers
+    for k, v := range c {
+        defaults[k] = v
+    }
+
+    // setting all default headers that are not included in the config
+    // file under the "default" category
+    for _, hn := range DefaultHeaderNames {
+        if _, ok := c[hn]; ok {
+            continue
+        }
+        hv := ParseDefaultHeaders(hn)
+        if hv != "" {
+            defaults[hn] = hv
+        }
+    }
+
+    return defaults
+}
+
+func ParseCustomResponseHeaders(r interface{}) (map[string]map[string]string, error) {
+    if !isValidListDict(r) {
+        return nil, fmt.Errorf("response headers were not configured correctly. please make sure they're in a map")
+    }
+
+    customResponseHeader := r.([]map[string]interface{})
+    h := make(map[string]map[string]string)
+
+    for _, crh := range customResponseHeader {
+        for sc, rh := range crh {
```

Comment:
Short var names are ok when it's obvious from local context what the thing refers to.  I don't think it's obvious here, consider using instead `statusCode, header := range crh`.

Reply:
I am struggling with these naming pattern in GO, and I have not figured out yet how to properly use them. I have seen in go that even for very long functions, these short var names are used which make it hard to search what it is. Appreciate the explanation. I will try to follow that!